Make RP experience a global setting

Meorawr opened the pull request:
Logically - it makes no sense for the RP experience field to be something that's profile-specific. Beginner users who want to elect into displaying this status on their profiles need to remember to set it up manually on each individual one, as it always defaults to experienced.

This changeset makes it such that the RP experience field is now stored as a configuration variable (not dissimilar to default profile IDs).

All logic for accessing and mutating the XP field has been relocated to the Player class accordingly. The XP field is no longer transmitted as part of "character" data but is now part of the vernum exchange, similar to the account trial information flag.

The XP status on player-owned profiles is set to nil with a migration to prevent us unnecessarily sending out stale data to older clients. We don't clear it on stored register profiles; the expectation is as people update the XP field will clear itself out progressively.

One downside to this is that the XP field as part of this change will now effectively reset for all users. It'd be a bit problematic to guess which experience level is correct if there's a mixture of profiles, so it's probably best to not do so and instead just make a note in the changelog. This does however give us a chance to look at resolving #178 by adding a fourth option and setting a new default that's not "Experienced".

Solanya commented on totalRP3/Core/Player.lua:
```diff
@@ -450,6 +450,25 @@ function CurrentUser:SetCustomRace(race)
+	local profileID = self:GetProfileID();
+
+	TRP3_API.configuration.setValue("roleplay_experience", experience);
+	character.v = TRP3_API.utils.math.incrementNumber(character.v or 1, 2);
```
I'm not sure there's a good way around it, but incrementing only the current profile means if someone sees you on another profile instead, they won't get the updated status. Probably unlikely to happen often anyway, just thinking out loud.

Meorawr replied:
Hmm, that's a good point. Not entirely opposed to just incrementing the versions across *all* profiles, but will think.

Meorawr commented:
Let's kill six birds with one bazooka and also take care of https://github.com/Total-RP/Total-RP-3/issues/178 while we're here, as the changes to XP to now be part of the vernum exchange is implicitly incompatible anyway.

Solanya commented:
Didn't notice there was a question.
Normal sounds weird to me right now, but I don't have a much better word to suggest, so we could just use that as default.

Meorawr commented:
Updated levels (and also corrected the field label itself):
![Wow_2024-07-03_23-32-37](https://github.com/Total-RP/Total-RP-3/assets/287102/05b6bf07-727e-44e8-92a4-6b5b0753a4a1)
![Wow_2024-07-03_23-32-40](https://github.com/Total-RP/Total-RP-3/assets/287102/ba5dfee0-0976-4091-a7eb-680f7116e305)
![Wow_2024-07-03_23-32-43](https://github.com/Total-RP/Total-RP-3/assets/287102/fc623e88-da5d-479a-a50b-fd4f3fd160aa)
![Wow_2024-07-03_23-32-46](https://github.com/Total-RP/Total-RP-3/assets/287102/0c98713d-c2cb-47c8-9d04-a579f4435eac)

New tooltip icons for retail:
![Wow_2024-07-03_23-25-40](https://github.com/Total-RP/Total-RP-3/assets/287102/d03c7256-8ef9-4acf-879d-54758d17ac81)
![Wow_2024-07-03_23-25-44](https://github.com/Total-RP/Total-RP-3/assets/287102/4fff3da9-6984-42ee-9303-e43de3a0f5fb)

Meorawr commented on totalRP3/Modules/Flyway/FlywayPatches.lua:
```diff
@@ -323,3 +322,23 @@ TRP3_API.flyway.patches["19"] = function()
+
+		if character then
+			character.XP = nil;
+			character.v = TRP3_API.utils.math.incrementNumber(character.v or 1, 2);
```
It both is and isn't - we want players to send an updated misc out with the XP field removed without having to wait until they touch their CU/CO fields to increment the version.
